### ISSUE TITLE
error[E0308] when compiling latest version

### ISSUE THREAD

**Issue by D33VIL** [stepancheg/grpc-rust#168]
As the title says, when compiling the latest version of grpc (v0.6.1) an error occurs.

```
Compiling grpc v0.6.1
error[E0308]: mismatched types
  --> /home/d33vil/.cargo/registry/src/github.com-1ecc6299db9ec823/grpc-0.6.1/src/protobuf.rs:21:62
   |
21 |         let mut is = CodedInputStream::from_carllerche_bytes(&buf);
   |                                                              ^^^^ expected struct `bytes::bytes::Bytes`, found struct `bytes::Bytes`
   |
   = note: expected type `&bytes::bytes::Bytes`
              found type `&bytes::Bytes`
note: Perhaps two different versions of crate `bytes` are being used?
  --> /home/d33vil/.cargo/registry/src/github.com-1ecc6299db9ec823/grpc-0.6.1/src/protobuf.rs:21:62
   |
21 |         let mut is = CodedInputStream::from_carllerche_bytes(&buf);
   |                                                              ^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
error: could not compile `grpc`.
```

**Comment by D33VIL** [comment]
As the error occurs for grpc and I use 
```toml
protobuf = { version = "2.8.1", features = ["with-bytes"] }
```
in my `Cargo.toml` for protobuf, I guess its `with-bytes` feature is not compatible with latest grpc.

**Comment by stepancheg** [comment]
I guess you can force bytes create version 0.4. Grpc-rust need to be fixed nevertheless.

**Comment by D33VIL** [comment]
Forcing the bytes crate version does't make a difference

**Comment by przygienda** [comment]
you have to specify protobuf="~2.8" now to make it work.

**Comment by stepancheg** [comment]
Published version 0.6.2 where versions are properly pinned (to rust-protobuf 2.8 and bytes 0.4).

Now working on upgrade to tokio 0.3 (rust-http2 crate source is migrated, now working on grpc-rust).
